core/src/Kokkos_Tuners.hpp

```diff
@@ -470,6 +485,79 @@ class TeamSizeTuner {
  private:
 };
 
+namespace Impl {
+template <typename Container>
+void fill_tile(Container& cont, int tile_size);
+
+template <typename T>
+void fill_tile(std::vector<T>& cont, int tile_size) {
+  for (int x = 1; x < tile_size; x *= 2) {
+    cont.push_back(x);
+  }
+}
+template <typename T, typename Mapped>
+void fill_tile(std::map<T, Mapped>& cont, int tile_size) {
+  for (int x = 1; x < tile_size; x *= 2) {
+    Mapped inner;
+    fill_tile(inner, tile_size / x);
+    cont[x] = inner;
+  }
+}
+}  // namespace Impl
+
+template <int MDRangeRank>
+struct MDRangeTuner {
+ private:
+  static constexpr const int Rank       = MDRangeRank;
+  static constexpr const int max_slices = 15;
+  using SpaceDescription =
+      typename Impl::n_dimensional_sparse_structure<int, Rank>::type;
+  using TunerType =
+      decltype(make_multidimensional_sparse_tuning_problem<max_slices>(
+          std::declval<SpaceDescription>(),
+          std::declval<std::vector<std::string>>()));
+  TunerType tuner;
+
+ public:
+  MDRangeTuner()        = default;
+  MDRangeTuner& operator=(const MDRangeTuner& other) = default;
+  MDRangeTuner(const MDRangeTuner& other)            = default;
+  MDRangeTuner& operator=(MDRangeTuner&& other) = default;
+  MDRangeTuner(MDRangeTuner&& other)            = default;
```

Comment:
Why not obey the rule of zero if we can.
```suggestion
```

Reply:
Good call